api/leaderworkerset/v1/leaderworkerset_types.go

```diff
@@ -85,6 +85,11 @@ type LeaderWorkerSetSpec struct {
 	// RolloutStrategy defines the strategy that will be applied to update replicas
 	// when a revision is made to the leaderWorkerTemplate.
 	RolloutStrategy RolloutStrategy `json:"rolloutStrategy"`
+
+	// PublishNotReadyAddresses indicates that any agent
+	// which deals with endpoints for this Service should disregard any indications of ready/not-ready.
+	// +kubebuilder:default=true
+	PublishNotReadyAddresses bool `json:"publishNotReadyAddresses"`
```

Comment:
hi @kannon92 As discussed with @ahg-g  I think we would make it enabled by default for now.

Reply:
Do you mean that you don't want anyone to set it to false?

Reply:
You may have commented on the Network API. I realized that adding that made it more complicated than it needed to be. I added a default setting to main LWS spec and I default to true in this case.

Reply:
yes for now, we could add the API field later when there is an ask. wdyt

Reply:
the reason for not adding a new field for this is bc we can't remove it later when we found out that exposing a new field is not necessary

Reply:
+1; lets not expose an api field for now until we get a case where we need to make it configurai

Reply:
sounds good. I updated the PR.